ec2/spark_ec2.py

```diff
@@ -349,6 +351,16 @@ def launch_cluster(conn, opts, cluster_name):
     if opts.identity_file is None:
         print >> stderr, "ERROR: Must provide an identity file (-i) for ssh connections."
         sys.exit(1)
+
+    if not os.path.exists(opts.identity_file):
+        print >> stderr, "ERROR: The identity file '{f}' doesn't exist.".format(f=opts.identity_file)
+        sys.exit(1)
+
+    file_mode = os.stat(opts.identity_file).st_mode
+    if not (file_mode & os.stat.S_IRUSR):
```

Comment:
Assuming we fix the errors here, won't this logic accept an identity file that is readable by others? As long as the owner can also read it, it looks like this check will accept the file, which is incorrect. Am I interpreting this correctly?